middlewares/campaigns.js

```diff
@@ -33,14 +35,32 @@ exports.registerUser = async (req, res) => {
             transaction: t
         });
 
+        // TODO: add uniqueness check; transliterate gsuiteEmail in case of umlauts or other special characters
+        const gsuiteEmail = req.body.first_name + '.' + req.body.last_name + '@' + constants.GSUITE_DOMAIN;
+        const payload = {
+            primaryEmail: gsuiteEmail.toLowerCase().replace(' ', ''),
+            name: {
+                givenName: req.body.first_name,
+                familyName: req.body.last_name
+            },
+            secondaryEmail: req.body.email,
+            password: crypto.createHash('sha1').update(JSON.stringify(req.body.password)).digest('hex'),
```

Comment:
@linuxbandit as per https://stackoverflow.com/questions/24925304/in-googles-directory-api-what-hash-format-is-required-when-crypt-is-provided we can also use SHA-512, can you try that in GSuiteWrapper?

Reply:
@serge1peshcoff this is what needs to be changed in GSuiteWrapper

Reply:
ah okay, haven't seen this comment somehow

Reply:
That's because we use the same function in two places. So it's good if we move it to a middleware where we can just define it once and call it twice

Reply:
agree 100%